### PR TITLE
Account Recovery: Add the reset password email form page

### PR DESCRIPTION
**Pull request by southp**
## Summary
image
This PR adds the page a user would see once the password reset link has been sent successfully. Note that the `email` is temporarily commented and replaced by a placeholder just for now.

## Test Plan
* Open an incognito window and make sure you are logged out.
* Visit http://calypso.localhost:3000/account-recovery/reset-password/email-form and make sure the above page is properly shown.
* The **You didn't receive it** link won't do anything for now.

### CODE REVIEW

**Comment by folletto** [comment]
Design wise it's ok. 👍  

The copy has been already reviewed? If not, maybe it's worth adding the "Needs Copy Review" tag to this PR. :)

**Comment by mikedang** [comment]
I like @omarjackman's language for the email, but would swap out the word "gotten" for "received":

> We've sent an email to test@example.com with a link to complete your password reset. 
> 
> If you haven't received it, please check your spam or junk folder.
> 
> [Resend reset link]

**Comment by southp** [comment]
@mikedang I've updated the wording. It now looks like this:
image

I haven't changed the "You didn't receive it?" though, since the original intention is not to allow users to trigger resending there. Here is the flow chart for quick reference: https://cloudup.com/cC9EfchsqDg . We are now at the step 5A. When a user clicks on that link, we would like to direct the user back step 4, so that one can reselect the other options or simply try again.  

In that context, does the sentence need to be adjusted?

cc @evilluendas in case I'm mistaken :)

File: client/account-recovery/reset-password/reset-password-email-form/index.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		translate,
+	} = props;
+
+	return (
+		<div className="reset-password-email-form">
+			<h2 className="reset-password-email-form__title">
+				{ translate( 'Reset your password' ) }
+			</h2>
+			<Card>
+			<p>
```
Comment (southp) [inline client/account-recovery/reset-password/reset-password-email-form/index.jsx]: Added by 144cdfca3c6cc6803e70780a0e8f75fef81fe113

File: client/account-recovery/reset-password/reset-password-email-form/index.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+				{ translate(
+					'We have sent an email to {{strong}}%(email)s{{/strong}} with a reset link. ' +
+					'Click on it to reset your password and recover access to your account.',
+					{
+						components: { strong: <strong /> },
+						// TODO: it will become args: { email } when the real data got wired in the up-coming PR.
+						// args: { email }
```
Comment (southp) [inline client/account-recovery/reset-password/reset-password-email-form/index.jsx]: Ah, nice tip :)
The default prop is now added by f3d6a75f94984b3d06eba1666122b37dd3947c97

**Comment by evilluendas** [comment]
I just realized that maybe we should show "We've sent an email to your main email address [etc.]" instead of displaying the actual address since we're hiding it from the previous screen for privacy and security reasons.

A middle ground (and what we had on the mockups, if I remember correctly) is some sort of obscured address (t**t@example.com for example) so you know where to look for the email, but it's not immediately obvious if you're trying to abuse the system.

**Comment by southp** [comment]
@evilluendas Sorry for the confusion. While the placeholder is `test@example.com`, the actual data coming from the server will be obscured. Thus, in the end it will be something like `t***@*****e.com`. As the placeholder will be removed soon in the future PR, I tend to leave it as it is.

Does that make sense to you? :)

**Comment by southp** [comment]
Also, by 688f9fe294ee7605933cfef1ea2dd9eadc4fab10 it now looks like this:
image
to have a consistent look with #11695 and #11732

File: client/account-recovery/reset-password/reset-password-email-form/index.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+					{
+						components: { strong: <strong /> },
+						args: { email }
+					}
+				) }
+			</p>
+			<p>
+				{ translate( "If you haven't received it, please check your spam or junk folder." ) }
+			</p>
+			<a href="#">{ translate( "You didn't receive it?" ) }</a>
```
Comment (jordwest) [inline client/account-recovery/reset-password/reset-password-email-form/index.jsx]: Could we change this to "Didn't receive it?" or "Try another method"?

Maybe it's just me, but somehow adding the "You" sounds a little accusatory :)

Comment (southp) [inline client/account-recovery/reset-password/reset-password-email-form/index.jsx]: Actually I think this is a good suggestion, especially given the fact that in its sibling SMS form we are saying "No SMS?". Updated by 3498e3c.

**Pull request merged**